### PR TITLE
Adding the WebView tracking feature

### PR DESCRIPTION
**Pull request by mariusc83**
### What does this PR do?

A brief description of the change being made with this pull request.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

**Review by xgouchet — CHANGES_REQUESTED** [review]
Nice work but we're missing a documentation on how to use the `DatadogEventBridge`. It'd be nice to also showcase this in the sample app.

File: dd-sdk-android/src/main/kotlin/com/datadog/android/rum/webview/RumWebChromeClient.kt [hunk 1]
```diff
@@ -20,6 +20,7 @@ import com.datadog.android.rum.RumErrorSource
  * Any console message will be forwarded to an internal [Logger], and errors
  * will be sent to the [GlobalRum] monitor as RUM Errors.
  */
+@Deprecated("You should use the DatadogEventBridge JavaScript interface instead.")
```
Comment (xgouchet) [inline dd-sdk-android/src/main/kotlin/com/datadog/android/rum/webview/RumWebChromeClient.kt:23]: Maybe we should add a snippet or indication here on _how to use_ the `DatadogEventBridge`.

File: dd-sdk-android/src/main/kotlin/com/datadog/android/rum/webview/RumWebViewClient.kt [hunk 2]
```diff
@@ -27,6 +27,7 @@ import com.datadog.android.rum.RumResourceKind
  * This will map the page loading, and webview errors into RUM Resource and
  * Error events respectively.
  */
+@Deprecated("You should use the DatadogEventBridge JavaScript interface instead.")
```
Comment (xgouchet) [inline dd-sdk-android/src/main/kotlin/com/datadog/android/rum/webview/RumWebViewClient.kt:30]: Maybe we should add a snippet or indication here on _how to use_ the `DatadogEventBridge`.

File: dd-sdk-android/src/main/kotlin/com/datadog/android/rum/webview/RumWebChromeClient.kt (hunk 1 above)
Comment (mariusc83) [reply dd-sdk-android/src/main/kotlin/com/datadog/android/rum/webview/RumWebChromeClient.kt:23]: I was thinking to add this in a separated task once we merge this in master. What do you think ?

Comment (xgouchet) [reply dd-sdk-android/src/main/kotlin/com/datadog/android/rum/webview/RumWebChromeClient.kt:23]: Well let's do it now, that we we won't forget it

File: dd-sdk-android/apiSurface [hunk 3]
```diff
@@ -109,6 +110,8 @@ data class com.datadog.android.core.configuration.Credentials
   constructor(String, String, String, String?, String? = null)
   companion object 
     const val NO_VARIANT: String
+object com.datadog.android.core.configuration.private
+const val VALID_IP_REGEX: String
```
Comment (0xnm) [inline dd-sdk-android/apiSurface:114]: strange, this doesn't look right

Comment (mariusc83) [reply dd-sdk-android/apiSurface:114]: Yup...I noticed that too, I will open a task to investigate this issue with the `apiSurface` generator.

**Review by xgouchet — APPROVED** [review]

**Review by 0xnm — APPROVED** [review]

**Pull request merged**